kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -63,18 +67,19 @@ public String channelDescriptor() {
     }
 
     /**
-     * Allocate a buffer with the given {@code initialCapacity}.
-     * The returned buffer will be released automatically
-     * TODO when?
-     * @param initialCapacity The initial capacity of the buffer.
-     * @return The allocated buffer.
+     * Creates a MemoryRecordsBuilder of the given capacity. The backing buffer will be deallocated when the request
+     * processing is completed
+     * @param initialCapacity The initial capacity of the backing buffer in bytes.
+     * @param compressionType The compression type.
+     * @param timestampType The timestamp type.
+     * @param baseOffset The bash offset.
+     * @return The created MemoryRecordsBuilder
      */
     @Override
-    public ByteBuf allocate(int initialCapacity) {
-        final ByteBuf buffer = channelContext.alloc().heapBuffer(initialCapacity);
+    public MemoryRecordsBuilder memoryRecordsBuilder(int initialCapacity, CompressionType compressionType, TimestampType timestampType, long baseOffset) {
```

Comment:
Is there milage in offering an overload with just `initialCapacity` and defaulting the `compression` and `timestamp` types as well as the `baseOffset` ?

Reply:
Having another look I saw that the MemoryRecordsBuilder has a large primary constructor that takes a lot of args. I made an alternative PR #152 that makes the context return a kafka ByteBufferOutputStream instead so that kroxy and filter authors can be more in control of creating the MemoryRecordsBuilder.